runtime/domain.c

```diff
@@ -272,10 +243,18 @@ static void create_domain(uintnat initial_minor_heap_wsize) {
       goto init_major_gc_failure;
     }
 
-    if(caml_reallocate_minor_heap(initial_minor_heap_wsize) < 0) {
-      goto reallocate_minor_heap_failure;
+    /* setting young_limit and young_ptr to minor_heaps_base
+       to trigger minor_heaps reallocation on GC poll */
+    domain_state->young_start = (char*)caml_global_minor_heap_start;
```

Comment:
This function `create_domain` can execute in parallel with STW segments but the domain is not yet part of the STW. It will be brittle to read `caml_global_minor_heap_start` multiple times should we ever change how that variable is assigned. 
Is there any reason to read this variable at all? Can't we just set all the `young_*` values to a suitable `null` value and trigger the collection we need when the values are used?